### PR TITLE
Tool for auditing dangling kindID references

### LINKED ISSUE #7655: Audit baggage for missing kindID handles
## What is the Problem Being Solved?

It is very easy to code vats that are intended to be upgradeable/restartable but which actually are not, due to failure to properly store kindID handles in the vat's baggage, such that when restart is attempted, durable objects cannot be associated with their behavior.  While the attempt to (re)define behavior upon restart will fail due to the missing kindID handle, at the point where this happens it may be too late to do anything about it.  We would like to have some kind of checker that can be used in one or more of the vat's tests to detect the problem prior to the vat's implementation being unleashed on an unsuspecting world.

(Note that this problem is distinct from failure to actually store necessary data in the vat's persistent state at all, which is the upgrade/restart failure mode that people working on vat durability have been mostly preoccupied with.  _This_ issue is concerned with the case where data has been successfully stored but is unable to be accessed due to inability to instantiate representatives of the objects that contain it.)

This issue surfaced in the comments on issue #7578, wherein @warner and @FUDCo started to design a test mechanism but then realized we were addressing a different problem from the one that issue was concerned with.  For the historical record, the relevant comments from that issue will be reproduced below.

## Description of the Design

The principal API will be a single function `checkBaggage(vatID, db)`, where `vatID` is the vatID of the vat to be audited and `db` is a handle to the open SQLite database.  The function will walk the transitive closure of the durable collections (`ScalarBigMapStore` and `ScalarBigSetStore`) and durable virtual objects reachable from the baggage map store of the indicated vat, collecting sets of (a) all the kindHandles that are reachable and (b) all the virtual object kinds that are in use.  It will also scan the `vom.dkind` entries and collect a set of all the kind IDs that are extant in that vat's vatstore.  If there are kind IDs that are defined in the database but which are not reachable from the baggage, it will print a report on its findings to the console log and throw an exception.

The expectation is that one or more of the vat's tests will call this function (and assert that it doesn't throw) when they have reached a point where the vatstore is expected to contain all the state necessary to have a fully durable, restartable vat.

Having gone to the work of doing the object graph walk, it may also make sense to provide an option to print a report of what it found regardless of whether or not there was a problem.

We should also provide a tool that wraps this function in a standalone command line executable that is given the path to a SQLite database file and performs the above check on the indicated database's contents.

Additionally, we should extend the debug hooks returned by `makeSwingStore` to add a function that will return the direct handle to the open SQLite database, so that tests that are using ephemeral in-memory databases (i.e., most of them) can provide the database to the checker (this is a feature that would have been useful in the past for other kinds of testing operations as well, so seems likely to be generally useful in the future too).

### PR DESCRIPTION
**Pull request by FUDCo**
Closes #7655

### CODE REVIEW

**Comment by FUDCo** [comment]
I'm now thinking that the files that are in `packages/SwingSet/tools` and `packages/SwingSet/misc-tools` should be somewhere in `packages/swing-store` (though not entire sure where in there).  This would also get rid of the change to `SwingSet/package.json`.

File: packages/SwingSet/package.json [hunk 1]
```diff
@@ -52,6 +52,7 @@
     "@endo/promise-kit": "^0.2.56",
     "@endo/ses-ava": "^0.2.40",
     "@endo/zip": "^0.2.31",
+    "better-sqlite3": "^8.2.0",
```
Comment (warner) [inline packages/SwingSet/package.json:55]: can we make this a devDependency? I feel like I've managed this in `misc-tools/` scripts before (like extract-transcript?).

Comment (warner) [reply packages/SwingSet/package.json:55]: oh, maybe the fact that it's in `tools/` is why it needs to be a real `dependency`

**Review by warner — APPROVED** [review]
sounds good.. I didn't look at it too closely, but the stakes are low if we missed something

**Comment by warner** [comment]
As discussed, maybe this wants to live in `packages/swingset-liveslots/` rather than in the kernel. OTOH it's kernel clients who should be using it, and they don't care that it really talks about data structures managed by liveslots.

**Pull request merged**